contracts/aggregator/polygon/flashloan/main.sol

```diff
@@ -184,6 +251,37 @@ contract FlashAggregatorPolygon is Helper {
         balancerLending.flashLoan(InstaFlashReceiverInterface(address(this)), wethTokenList_, wethAmountList_, data_);
     }
 
+    /**
+     * @dev Middle function for route 9.
+     * @notice Middle function for route 9.
+     * @param _tokens list of token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _data extra data passed.
+    */
+    function routeInterop(address[] memory _tokens, uint256[] memory _amounts, bytes memory _data) internal {
+        bytes memory data_ = abi.encode(8, _tokens, _amounts, msg.sender, _data);
+        dataHash = bytes32(keccak256(data_));
+        interopLending.initiateFlashLoan(address(this), _tokens, _amounts, data_);
+    }
+
+    /**
+     * @dev Middle function for route 9.
+     * @notice Middle function for route 9.
+     * @param _tokens list of token addresses for flashloan.
+     * @param _amounts list of amounts for the corresponding assets.
+     * @param _data extra data passed.
+     * @param _instaData extra data passed.
+    */
+    function routeInteropAave(address[] memory _tokens, uint256[] memory _amounts, bytes memory _data, bytes memory _instaData) internal {
+        (address[] memory tokens_, uint256[] memory amounts_) = abi.decode(
+            _instaData,
+            (address[], uint256[])
+        );
+        bytes memory data_ = abi.encode(9, _tokens, _amounts, msg.sender, _data);
+        dataHash = bytes32(keccak256(data_));
```

Comment:
@Vaibhav20201 Are we using this `dataHash` somewhere?

Reply:
Its just an extra layer of security, we hash it again and check it in the modifier.

Reply:
But we are not using this `dataHash` anywhere in this function block, right?

Reply:
Not used in this function.
It is used here
https://github.com/Instadapp/flashloan-aggregator/blob/5bf069bcd4ee6df1ef32a74eea0b4e1fe2f970b8/contracts/aggregator/avalanche/flashloan/main.sol#L88 in the modifier.
Checkout this modifier:-
https://github.com/Instadapp/flashloan-aggregator/blob/5bf069bcd4ee6df1ef32a74eea0b4e1fe2f970b8/contracts/aggregator/avalanche/flashloan/helpers.sol#L250

Reply:
Yeah! I get that but in `routeInteropAave` function block, where are we `dataHash` variable? 
![Uploading image.png…]()

Reply:
We dont use it in that function. We just set it there. And then we check it in the fallback function of the respective route.